### ISSUE TITLE
Destroy then initialize does not respect validation on consecutive submissions

### ISSUE THREAD

**Issue by TrevorHinesley** [erikras/redux-form#2934]
I have an edit profile page with a simple name/email/password form. Let's say the user changes their name, then hits save. If they then change their name again and hit save, it won't validate anything (i.e., they can submit a blank name which is supposed to be required, or a password under the minimum length, etc.)

I'm handling side effects using redux-saga, so I call startSubmit, stopSubmit, and destroy separately.

However, after this form is destroyed, I do an initialize with the server's responded values to re-initialize it.

For some reason, destroy gets rid of validation. How do I initialize after the destroy, and pass validation to it? My handleSubmit (which is declared via onSubmit in the reduxForm method in the component) still works fine, but validation does not.

Again, validation works fine on first submission. But after destroy, it doesn't work any longer.

**Comment by TrevorHinesley** [comment]
Also, this works if I remove step 6. Just wondering why destroy does this, and how one would re-initialize manually, keeping validation after a destroy.

**Comment by kvedananda** [comment]
I think I'm having a similar issue. I use the same form to handle creating and editing. Validation works fine at first but if I create something then try to edit it the validation doesn't work correctly. It actually does validate the form and correctly shows the errors but I'm still able to submit my form.

**Comment by TrevorHinesley** [comment]
@kvedananda try removing destroy if you're using destroy. Like I said, that fixed it for me. Not sure if it's right or not, but it works.

**Comment by kvedananda** [comment]
@TrevorHinesley In my case the form is on two different routes/pages so it seems it's the same problem as yours but in my case I don't think I can prevent a destroy. Since validation works fine I ended up disabling the submit button if the syncErrors object was present in the form state and this at least prevents an invalid submission.

**Comment by TrevorHinesley** [comment]
I could be wrong, but I don't believe the route changing should affect whether or not the form is destroyed, as that's just a typical unmount.

**Comment by kvedananda** [comment]
Forms are destroyed on unmount. I see the destroy event in the console when switching routes.

Did some hunting around and seems this is enabled by default but there is an option to change this.  http://redux-form.com/6.7.0/docs/api/ReduxForm.md/#-destroyonunmount-boolean-optional-

May have to try this out...

**Comment by TrevorHinesley** [comment]
That's correct. I forgot I had that set in my form because it's a wizard. But yes, you'll need to set that option to false!

**Comment by BrendonEarl** [comment]
Any update on this? I'm experience the same issue on version 7.1.2 using React 15.6.2 and Redux 3.7.2. Where after calling destroy, the form will submit regardless of if there are any `syncErrors` sitting in the form state

**Comment by erikras** [comment]
For all of you using `redux-saga`, perhaps [check out this tool I just built](https://medium.com/@erikras/getting-a-promise-from-redux-actions-d92787894368) to convert actions into a `Promise` for libraries like `redux-form`.
